src/layers/core/grid-layer/grid-layer.js

```diff
@@ -54,25 +56,62 @@ function _needsReProjectPoints(oldProps, props) {
   return oldProps.cellSize !== props.cellSize;
 }
 
+function _percentileChanged(oldProps, props) {
+  return oldProps.lowerPercentile !== props.lowerPercentile ||
+    oldProps.upperPercentile !== props.upperPercentile;
+}
+
+function _needsReSortBins(oldProps, props) {
+  return oldProps.getColorValue !== props.getColorValue;
```

Comment:
I added comment to layer doc.
If people pass in a new function every time, the layer is just going to sort the bins on every render, which won't result in error showing colors. But if we don't check the function change, and not re sort the bins when it actually changed. we have no way to detect whether some one pass in a new function to calculate color value.